### DISCUSSION TITLE
Vercel build use preview env by default, allow passing other environment

### DISCUSSION THREAD

**Discussion by FabienGreard** [vercel/vercel#8982] [Ideas]
👋 Hi ! 

context: We are doing something very similar to https://vercel.com/guides/how-can-i-use-github-actions-with-vercel. 

The only difference is that we use the development environment too. 

Looks like that `vercel build` could be improved.

If we follow the [doc](https://vercel.com/docs/cli/build).

Right now there is no way to specify something else than prod or preview environments.

Meaning doing ```vercel build``` will use the preview env by default. 

Meaning ```vercel pull --environment=development``` is not useful at all as the build command will always use the preview env. 

Today, to make it works we are using custom env by manually adding those in the ci as passing down system env is allowed. 

I suppose using [vercel env pull .env](https://vercel.com/docs/cli/env) could also work.

Sorry if this was already pointed out but couldn't find it.

**Comment by amyegan** [comment]
Thanks for sharing this! At the moment, you can use [`vercel dev`](https://vercel.com/docs/cli/dev) to replicate Vercel's deployment in your development environment. If you're using a framework and your framework's Development Command already provides all the local dev features you need, it's typically better to use that and pull in your development environment variables with [`vercel env pull [file]`](https://vercel.com/docs/cli/env#extended-usage). 

If those possibilities don't handle what you need, please tell me more about your workflow. I want to make sure the engineering team has the right information to be able to explore this idea!

**Reply by FabienGreard** [reply]
thanks for the reply ! So the thing is `vercel dev` is made for local development, here what we want to do is to use the development environment for deploying which seems to be impossible unless modifying the system env, where this could be super simple if we had a way to specify it while doing a build such as `vercel build --prod` but for dev. something like `vercel build --environment=<production|preview|development>` would be super neat ! This is exactly how [vercel pull](https://vercel.com/docs/cli/pull) works.

**Reply by amyegan** [reply]
Excellent feedback. I'll share this with the team. Thanks, Fabien!
